src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolutionResult.cs

```diff
@@ -476,7 +476,10 @@ private static ThreeState TryGetBestResult(ArrayBuilder<MemberResolutionResult<T
                     && !isMethodGroupConversion
                     // Function pointer type symbols don't have named parameters, so we just want to report a general mismatched parameter
                     // count instead of name errors.
-                    && !(firstSupported.Member is FunctionPointerMethodSymbol _))
+                    // The only case we want this block to be entered for function pointers is if it's called with a named argument.
```

Comment:
If function pointers are **always** allowed to enter this block, we will start to get strange error messages like:

> error CS7036: There is no argument given that corresponds to the required formal parameter '' of 'delegate*<string, void>'

which happens when on MemberResolutionKind.RequiredParameterMissing.

So my change only allowed function pointers to enter here for NoCorrespondingNamedParameter only.